test/jdk/javax/swing/JFileChooser/6798062/bug6798062.java

```diff
@@ -21,54 +21,87 @@
-            java.desktop/sun.awt.shell
-   @run applet/manual=done bug6798062.html
-*/
+/*
+ * @test
+ * @bug 6798062
+ * @requires (os.family == "windows")
+ * @summary Memory Leak on using getFiles of FileSystemView
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @modules java.desktop/sun.awt
+ *          java.desktop/sun.awt.shell
+ * @run main/manual bug6798062
+ */
 
 import sun.awt.OSInfo;
 import sun.awt.shell.ShellFolder;
 
-import javax.swing.*;
+import java.awt.Dimension;
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
-import java.awt.*;
 import java.io.File;
 import java.io.FileNotFoundException;
-
-public class bug6798062 extends JApplet {
-
-    private final JSlider slider = new JSlider(0, 100);
-
-    private final JTextField tfLink = new JTextField();
-
-    private final JButton btnStart = new JButton("Start");
-
-    private final JButton btnStop = new JButton("Stop");
-
-    private final JButton btnGC = new JButton("Run System.gc()");
+import javax.swing.BoxLayout;
+import javax.swing.JButton;
+import javax.swing.JComponent;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JSlider;
+import javax.swing.JTextField;
+import javax.swing.SwingUtilities;
+
+public class bug6798062 {
+
+    private static final String instructionsText = """
+            The test is suitable only for Windows.
+
+            1. Create a link
+            2. Copy path to the link into TextField
+            3. Run the Windows Task Manager. Select the Processes tab and find the java process
+            4. Press the Start button in the test window
+            5. Wait several minutes and observe in the Windows Task Manager
+               that Memory Usage of java process is not increasing
+            If memory usage is increasing, click Fail else click Pass .  """;
+
+    private static JSlider slider;
+    private static JFrame frame;
+    private static JTextField tfLink;
+    private static JButton btnStart;
+    private static JButton btnStop;
+    private static JButton btnGC;
 
     private ShellFolder folder;
-
     private Thread thread;
 
-    public static void main(String[] args) {
-        JFrame frame = new JFrame("bug6798062");
-
-        frame.setSize(400, 300);
-        frame.setLocationRelativeTo(null);
-        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
-        frame.add(new bug6798062().initialize());
-
-        frame.setVisible(true);
-    }
-
-    public void init() {
-        add(initialize());
+    public static void main(String[] args) throws Exception {
+         PassFailJFrame passFailJFrame = new PassFailJFrame.Builder()
+                .title("JFileChooser Instructions")
+                .instructions(instructionsText)
+                .testTimeOut(5)
+                .rows(10)
+                .columns(35)
+                .build();
+
+        SwingUtilities.invokeAndWait(() -> {
```

Comment:
I suggest using the `testUI` method of the builder which accepts a lambda expression that returns a frame. Calling the passed method reference or lambda expression on EDT, registering the frame with `PassFailJFrame` and its positioning is handled automatically.

Reply:
ok..updated...